server/graphql/v2/query/ContributorQuery.ts

```diff
@@ -0,0 +1,68 @@
+import assert from 'assert';
+
+import { GraphQLNonNull } from 'graphql';
+
+import { sequelize } from '../../../models';
+import { BadRequest } from '../../errors';
+import { fetchAccountWithReference, GraphQLAccountReferenceInput } from '../input/AccountReferenceInput';
+import { GraphQLContributor } from '../object/Contributor';
+
+const ContributorQuery = {
+  description: 'Get Contributors grouped by their profiles',
+  type: new GraphQLNonNull(GraphQLContributor),
+  args: {
+    account: {
+      type: GraphQLAccountReferenceInput,
+      description: 'Contributor Account reference',
+    },
+    host: {
+      type: GraphQLAccountReferenceInput,
+      description: 'Context host to fetch the contributor from',
+    },
+  },
+  async resolve(_: void, args, req: Express.Request) {
+    assert(args.account && args.host, 'You must provide either an account or a host to fetch the contributors');
+    const account = await fetchAccountWithReference(args.account, { throwIfMissing: true });
+    const host = await fetchAccountWithReference(args.host, { throwIfMissing: true });
+    if (!req.remoteUser?.isAdminOfCollective(host)) {
+      throw new BadRequest('Only admins can lookup for members using the "host" argument');
+    }
```

Comment:
There's a lack of consistency in what should be required:
- Parameters are optional 
- `assert` enforces both arguments are set (but the message says "Or")
- Direct fetch with arg check also assumes both should be

Reply:
We have an issue here. This host contributor we're returning here is a subset of the existing "Contributor" object.
I'll move these functions and DRY them, but they won't be compatible between existing library queries and this new query.